src/resmom/mom_hook_func.c

```diff
@@ -3075,6 +3088,437 @@ record_job_last_hook_executed(unsigned int hook_event,
 	fclose(fp);
 }
 
+/**
+ * @brief
+ * This function runs after execution of a single hook and processes
+ * the results from hook execution. If hook is backgrounded,on
+ * successful execution, a new task will be created to run the next
+ * hook script and if there was an error (the hook process returned a non-zero exit
+ * status) it does not create the new task for the next hook script.
+ * 
+ * @param[in] 	ptask - the work task.
+ *
+ * @return 1 a hook accepted
+ * @return 0 a hook rejected
+ * @return -1 an internal error occurred
+ */
+static int
+post_run_hook(struct work_task *ptask)
+{
+
+	int    accept_flag = 1;
+	int    reject_flag = 0;
+	int    reject_rerunjob = 0;
+	int    reject_deletejob = 0;
+	int    reboot_flag = 0;
+	int    log_type = 0;
+	int    log_class = 0;
+	int    hook_error_flag = 0;
+	int    *reject_errcode = NULL;
+	int    wstat = 0;
+	char   *log_id = NULL;
+	char   reject_msg[HOOK_MSG_SIZE+1] = {'\0'};
+	char   reboot_cmd[HOOK_BUF_SIZE]  = {'\0'};
+	char   hook_outfile[MAXPATHLEN+1] = {'\0'};
+	pid_t  mypid = 0;
+	hook   *phook	= NULL;
+	mom_hook_input_t *hook_input 	= NULL;
+	job *pjob	= NULL;
+	struct work_task *new_task;
+	pbs_list_head	vnl_changes;
+	mom_process_hooks_params_t *php;
+
+	if (ptask == NULL) {
+		log_err(-1, __func__, "missing ptask argument to event");
+		return -1;
+	}
+
+	if ((phook = (hook *)ptask->wt_parm1) == NULL) {
+		log_err(-1, __func__, "missing hook phook argument to event");
+		return -1;
+	}
+
+	if ((php = (mom_process_hooks_params_t *) ptask->wt_parm2) == NULL) {
+		log_err(-1, __func__, "missing hook params argument to event");
+		return -1;
+	}
+
+	if (php->hook_event == HOOK_EVENT_EXECHOST_PERIODIC) {
+		post_periodic_hook(ptask);
+		return 1;
+	}
+
+	if ((hook_input = (mom_hook_input_t *) php->hook_input) == NULL) {
+		log_err(-1, __func__, "missing input argument to event");
+		return -1;
+	}
+	
+	pjob = (job *)hook_input->pjob;
+	CLEAR_HEAD(vnl_changes);
+
+	if ((php->hook_event == HOOK_EVENT_EXECHOST_PERIODIC) ||
+		(php->hook_event == HOOK_EVENT_EXECHOST_STARTUP)) {
+		log_id = phook->hook_name;
+		log_type = PBSEVENT_DEBUG2;
+		log_class = PBS_EVENTCLASS_HOOK;
+	} else {
+		log_id = pjob->ji_qs.ji_jobid;
+		log_type = PBSEVENT_JOB;
+		log_class = PBS_EVENTCLASS_JOB;
+	}
+
+	mypid = php->child;
```

Comment:
setting of mypid to php->child is not effective for background hooks as we set it to ptask->wt_event on line 3173.
Also, could you please explain how having the pid in the log will help?

Reply:
We are forming the hook out file name using child's PID in the [post_run_hook](https://github.com/PBSPro/pbspro/pull/915/files#diff-f000d05f102c29b1985e578963befa7aR3217) function.
We are not using PID in logs in post_run_hook function.